obs2ioda-v2/src/f_c_string_t_mod.f90

```diff
@@ -0,0 +1,54 @@
+module f_c_string_t_mod
+    use iso_c_binding, only : c_loc, c_ptr, c_null_char, c_char, c_f_pointer, c_null_ptr
+    use f_c_string_i_mod, only : strlen
+    implicit none
+
+    type :: f_c_string_t
+        ! Allocatable Fortran string
+        character(len = :), allocatable :: f_string
+        ! Allocatable C-compatible null-terminated string
+        character(len = :, kind = c_char), allocatable :: fc_string
+        ! C pointer to the null-terminated string
+        type(c_ptr) :: c_string
+        ! Length of the Fortran string
+        integer :: n = -1
+
+    contains
+        ! Type-bound procedures
+        procedure :: to_c => to_c
+        procedure :: to_f => to_f
+    end type f_c_string_t
```

Comment:
How does someone using the `f_c_string_t` type ensure that memory is freed after an instance is no longer needed? One possibility that we discussed would be to use a `final` method.

Reply:
I’ve addressed this in my latest commit, which includes a `final` method for the `f_c_string_t type`. The `final` method automatically handles memory cleanup when an instance of the type goes out of scope or is explicitly deallocated. You can review the implementation in detail [here](https://github.com/NCAR/obs2ioda/commit/5d5164878889dc2bcc0f5fcd295e52c7b97d916d#diff-073eb5a19136a2de0a6aa2bfa4893e4f049d6ad08bb5570725d71c6b78894bd8R48)